cdp_scrapers/instances/houston.py

```diff
@@ -0,0 +1,283 @@
+from bs4 import BeautifulSoup, NavigableString, Tag
+from cdp_backend.pipeline import ingestion_models
+from cdp_backend.pipeline.ingestion_models import EventIngestionModel
+
+from cdp_scrapers.scraper_utils import IngestionModelScraper
+
+log = logging.getLogger(__name__)
+
+
+class HoustonScraper(IngestionModelScraper):
+    def __init__(self):
+        super().__init__(timezone="America/Chicago")
+
+    def remove_extra_type(self, element: Union[Tag, NavigableString, None]) -> Tag:
+        """
+        Remove types that are not useful
+
+        Parameter:
+        ----------------
+        event: Union[Tag, NavigableString, None]
+            All elements in the page that we want to scrape
+
+        Returns:
+        ----------------
+        Tag
+            Same elements as received, assuming the elements are not null
+        """
+        if isinstance(element, NavigableString) or element is None:
+            raise ValueError(f"Wrong Type {type(element)}")
+        return element
+
+    def get_body_name(self, event: Union[Tag, NavigableString, None]) -> str:
+        """
+        Get the body name for an event
+
+        Parameter:
+        ----------------
+        event: Union[Tag, NavigableString, None]
+            All elements in the page that we want to scrape
+
+        Returns:
+        ----------------
+        str
+            The body name
+        """
+        log.info("start get body name")
+        event = self.remove_extra_type(event)
+        bodyTable = event.find_all("table")[1].find("table")
+        if "CITY COUNCIL" in bodyTable.text:
+            return "City Council"
+        else:
+            return bodyTable.find_all("span")[3].text.title()
+
+    def get_event_Minutes_Item(
+        self,
+        event: Union[Tag, NavigableString, None],
+    ) -> List[ingestion_models.EventMinutesItem]:
+        """
+        Parse the page and gather the event minute items
+
+        Parameter:
+        ----------------
+        event: Union[Tag, NavigableString, None]
+            All elements in the page that we want to scrape
+
+        Returns:
+        ----------------
+        List[ingestion_models.EventMinutesItem]
+            All the event minute items gathered from the event on the page
+        """
+        log.info("start get items")
+        event_minutes_items = []
+        all_items = self.remove_extra_type(event).find_all("td", {"class": "style4"})
+        for item in all_items:
+            name = ""
+
+            for i in item.stripped_strings:
+                name = name + " " + repr(i).replace("'", "")
+
+            if name is not None and name != "":
+                event_minutes_items.append(
+                    ingestion_models.EventMinutesItem(
+                        minutes_item=ingestion_models.MinutesItem(name.strip())
+                    )
+                )
+
+        return event_minutes_items
+
+    # Big Functions
+    def get_diff_yearid(self, time: datetime) -> str:
```

Comment:
`time` can be confusing, e.g. `datetime.time` or `time` modules. Would be better to use a less common word for this variable name.

Reply:
Done